mmcv/ops/csrc/pytorch/cpu/bbox_overlaps_cpu.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright(c) OpenMMLab.All rights reserved.
+#include "pytorch_cpp_helper.hpp"
+#include "pytorch_device_registry.hpp"
+#define HOST_DEVICE
```

Comment:
What's this for?

Reply:
I'm sorry, maybe I copied and pasted it up by mistake

Reply:
I improved my test case. cuda didn't pass the semi-precision before, which should be because my range was too large and I added the previous smaller integer, resulting in the loss of precision. Therefore, I reduced the range and increased the addition number
```
        b1 = torch.tensor([[10.0+i, 10.0+i, 30.0+i, 30.0+i] for i in range(1000)]).to(device).type(dtype)
        b2 = torch.tensor([[20.0+i, 20.0+i, 40.0+i, 40.0+i] for i in range(1000)]).to(device).type(dtype)
        should_output = np.array([1/7]*1000)
        out = bbox_overlaps(b1, b2, aligned=True)
        assert np.allclose(out.cpu().numpy(), should_output, 1e-2)
```

cpu version
![截图 2023-01-14 10-18-47](https://user-images.githubusercontent.com/30740917/212446783-23e41969-630e-4b67-8653-f045e2fa3e76.png)

pure python version
![截图 2023-01-14 10-18-13](https://user-images.githubusercontent.com/30740917/212446790-c3b32bd7-9426-44cb-9718-ac0cf3a22596.png)


In fact, due to the volatility of benchmark results, there was not much difference between the two in my multiple tests, and the cpu version had a slight advantage over the python version

Reply:
cool